### DISCUSSION TITLE
How to use `AddBearerToken` without Identity

### DISCUSSION THREAD

**Discussion by UltraWelfare** [dotnet/aspnetcore#53795] [Q&A]
> The new bearer token authentication handler integrates seamlessly with ASP.NET Core’s built-in authentication system. It can be used standalone (without relying on ASP.NET Core identity). It supports issuing and validating tokens. As of Preview 6, it also supports refresh tokens. ASP.NET Core Identity uses the AddBearerToken extension to integrate the handler with identity.

However it isn't properly documented anywhere on MS Docs on how to use it without Identity.
After digging around the internet I found out you can use `Results.SignIn(claimsPrincipal)` which gives you the access token and refresh token.

I'm not completely sure on how to use the refresh token though. For a website I'd need to send it as an HttpOnly cookie however I'm not sure how I would grab this on the endpoint since the resulting type of `SignIn` is `IResult`..?

And then how would I create a `/refresh` endpoint to validate that it is correct?

Any help appreciated!

**Comment by paulblamire** [comment]
Also from digging around: 

https://github.com/dotnet/aspnetcore/blob/3f99d45b0b7d8f0427a3d98acc63098694613362/src/Identity/Core/src/IdentityApiEndpointRouteBuilderExtensions.cs#L122

That is the code that registers the refresh link if you use it WITH identity. So it will need to be very similar but doing your own thing instead of using signInManager

**Comment by RichardD012** [comment]
I don't see any way to validate the tokens. It seems that something is inherently keeping the tokens safe, as requests are validating the accessTokens but there isn't any documentation on the correct way to validate the refresh token.
